retrievalmarket/discovery/local.go

```diff
@@ -22,17 +19,48 @@ func NewLocal(ds datastore.Batching) *Local {
 }
 
 func (l *Local) AddPeer(cid cid.Cid, peer retrievalmarket.RetrievalPeer) error {
-	// TODO: allow multiple peers here
-	//  (implement an util for tracking map[thing][]otherThing, use in sectorBlockstore too)
-
-	log.Warn("Tracking multiple retrieval peers not implemented")
-
-	entry, err := cbor.DumpObject(peer)
+	key := dshelp.CidToDsKey(cid)
+	exists, err := l.ds.Has(key)
 	if err != nil {
 		return err
 	}
 
-	return l.ds.Put(dshelp.CidToDsKey(cid), entry)
+	var newRecord []byte
+
+	if !exists {
+		newRecord, err = cbor.DumpObject([]retrievalmarket.RetrievalPeer{peer})
+		if err != nil {
+			return err
+		}
+	} else {
+		entry, err := l.ds.Get(key)
+		if err != nil {
+			return err
+		}
+		var peerList []retrievalmarket.RetrievalPeer
+		if err = cbor.DecodeInto(entry, &peerList); err != nil {
+			return err
+		}
+		if hasPeer(peerList, peer) {
+			return nil
+		}
+		peerList = append(peerList, peer)
+		newRecord, err = cbor.DumpObject(peerList)
+		if err != nil {
+			return err
+		}
+	}
+
+	return l.ds.Put(key, newRecord)
+}
+
+func hasPeer(peerList []retrievalmarket.RetrievalPeer, peer retrievalmarket.RetrievalPeer) bool {
```

Comment:
I'm assuming these are going to be relatively short lists.

Reply:
yea shouldn't be too bad